File: cs50x/session2021/pset6/readability/readability.py
###
#-------------------------------------------------------------------------------
# readability.py
#-------------------------------------------------------------------------------
#
# Author:       Alwin Tareen
# Created:      Feb 13, 2021
# Execution:    python3 readability.py
# Check50:      check50 cs50/problems/2021/x/sentimental/readability
# Submit50:     submit50 cs50/problems/2021/x/sentimental/readability
#
# This program computes the approximate grade level needed to read some text.
#
##

import string
prose = input("Text: ")

letters = 0
words = 0
sentences = 0

for item in prose:
    if item in string.ascii_letters:
        letters += 1
    elif item == " ":
        words += 1
    elif item in ".!?":
        sentences += 1

words += 1

#print(f"{letters} letter(s)")
#print(f"{words} word(s)")
#print(f"{sentences} sentence(s)")

L = 100.0 * letters / words
S = 100.0 * sentences / words

index = 0.0588 * L - 0.296 * S - 15.8

grade = round(index)

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: cs50x/session2021/pset2/readability/readability.c
/**
|-------------------------------------------------------------------------------
| readability.c
|-------------------------------------------------------------------------------
|
| Author:       Alwin Tareen
| Created:      Jan 31, 2021
| Compilation:  make readability
| Execution:    ./readability
| Check50:      check50 cs50/problems/2021/x/readability
| Submit50:     submit50 cs50/problems/2021/x/readability
|
| This program determines the approximate grade level needed to comprehend some
| text.
|
*/

#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int letters = 0;
    int words = 0;
    int sentences = 0;
    const int BUFFER = 1024;
    char text[BUFFER];
    
    printf("Text: ");
    fgets(text, BUFFER, stdin);
    
    for (int i = 0, size = strlen(text); i < size; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        
        if (text[i] == ' ')
        {
            words++;
        }
        
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
        
    }
    words++;
    
    double L = letters * (100.0 / words);
    double S = sentences * (100.0 / words);
    
    double index = 0.0588 * L - 0.296 * S - 15.8;
    int result = (int) round(index);
    
    if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (result >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", result);
    }
    
    //printf("%d letters(s)\n", letters);
    //printf("%d word(s)\n", words);
    //printf("%d sentence(s)\n", sentences);
    
    return 0;
}

